plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/blobstore/AzureBlobContainer.java

```diff
@@ -72,14 +72,14 @@ public InputStream readBlob(String blobName) throws IOException {
         logger.trace("readBlob({})", blobName);
 
         if (!blobExists(blobName)) {
```

Comment:
@imotov : What I meant was that it didn't throw Exceptions during testing (e.g. when I tried deleting a blob that didn't exist).  Of course, this was using a mock, so I don't know if that would be the case in real life, but testing purposes, the existence check was necessary for it to pass.